src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/listview.controller.js

```diff
@@ -680,19 +680,29 @@ function listViewController($scope, $interpolate, $routeParams, $injector, $time
             if (isDate(value)) {
                 value = value.substring(0, value.length - 3);
             }
-
             if (e.nameExp) {
-                var newValue = e.nameExp({ value });
-                if (newValue && (newValue = newValue.trim())) {
-                    value = newValue;
-                }
+                value = resolveExpression(e, result, value, alias, 0);
             }
 
             // set what we've got on the result
             result[alias] = value;
         });
     }
 
+    function resolveExpression(e, result, value, alias, retestCount) {
+        var newValue = e.nameExp({ value });
+        if (newValue && newValue.startsWith('Loading...') && retestCount < 5) {
+            retestCount++;
+            $timeout(function () {
+              resolveExpression(e, result, value, alias, retestCount);
+            }, retestCount * 1000);
+        }
+        if (newValue && (newValue = newValue.trim())) {
+            result[alias] = newValue;
```

Comment:
Why does it need to set value here, which it also does in https://github.com/umbraco/Umbraco-CMS/pull/13086/files#diff-ca17a4d4a225dd1c3638ac33bb441693feb7dc372164667adebab614ec635d5dL692 ?

I think the function should just be responsible to resolving the value, but not updating property value.

Reply:
It's doing that so that it can update the resulting value when it's called recursively. Working on a better way to handle this though, as relying on timeout doesn't always work.

Reply:
New code is up now. The way it works is that it passes the interpolated function (e.nameExp) to the umb-table component. I then tweaked the umb-table html so that it uses that function if it has been set. Otherwise it uses the original behavior.